packages/components/src/Combobox/Combobox.test.tsx

```diff
@@ -251,12 +263,13 @@ describe("Combobox selected value", () => {
   it("has no selected option when a null selected value is passed", () => {
     const { getByText } = render(<ClearSelectionCombobox />);
 
-    fireEvent.click(getByText("Button"));
-
     const option = getByText("Bilbo Baggins");
-    expect(option).toHaveClass("selectedOption");
     const clearButton = getByText("Clear Selection");
+
+    expect(option).toHaveClass("selectedOption");
```

Comment:
had to shuffle around the order of interactions here

this `expect` where we check the class only works because the open state is being managed with visibility right now. if this isn't rendered it'll stop working which is perhaps not ideal

what was causing this test to fail was that we only opened the dialog to check, then cleared but never closed the dialog, which means we depend on some re-renders as `open` changes to get the correct values.  

you wouldn't be able to do this in real life. your tabs & focused would either be trapped in the dialog, or you click outside and it closes

well actually now that I say it out loud, you could do this if you gave the combobox an option to clear the selection which I think would be super weird and not how this component should be used at all. even still, you could do that - and you'd probably hit the bug this test was hitting

Reply:
nevermind, even though that's a weird thing to do it works without causing a bug

<img width="1506" alt="image" src="https://github.com/GetJobber/atlantis/assets/11843215/e7cf11b2-2e89-4a68-97a9-27a5d34637c4">

I'm willing to chalk it up to the test environment behaving differently than real life 🤷